source/fundamentals/query-data.txt

```diff
@@ -0,0 +1,194 @@
+
+You can use the ``Where()`` method to retrieve multiple entities from your
+collections. ``Where()`` returns all entities that match the search
+criteria.
+
+The following code uses the ``Where()`` method to find all planets that have the
+``hasRings`` field set to ``true`` and prints the planet names to the console.
+
+.. literalinclude:: /includes/code-examples/quick-reference.cs
+   :start-after: // start-find-many
+   :end-before: // end-find-many
+   :language: csharp
+   :dedent:
+
+To learn more about the ``Where()`` method, see the `.NET API documentation
+<https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.query.queryablemethods.where>`__.
+
+Find All Entities
+~~~~~~~~~~~~~~~~~
+
+You can get all entities from your database by using the ``ToList()`` method.
+The ``ToList()`` method will return all entities in the specified collection as
+a ``List``.
+
+The following code uses the ``ToList()`` method to return all entities in a
+``DBSet`` called ``Planets`` and prints their names to the console.
+
+.. code-block:: csharp
+
+   var allPlanets = db.Planets.ToList();
+
+   foreach (var p in allPlanets)
+    {
+        Console.WriteLine(p.name);
+    }
+
+To learn more about the ``ToList()`` method, see the `.NET API documentation
+<https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tolist>`__.
+
+Sort Entities
+-------------
+
+Use the ``OrderBy()`` method to specify an order in which to return entities
+from a query. ``OrderBy()`` sorts the elements in ascending order based on a
+specified sort criteria.
+
+The following code uses the ``OrderBy()`` method to find all planets and sort
+them by the value of the ``orderFromSun`` field in ascending order. It then
+prints the results to the console.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var planetList = db.Planets.OrderBy(p => p.orderFromSun).ToList();
+    
+      foreach (var p in planetList)
+      {
+         Console.WriteLine(p.name);
+      }
+
+   .. output::
+      :language: json
+      :visible: false
+
+      Mercury
+      Venus
+      Earth
+      Mars
+      Jupiter
+      Saturn
+      Uranus
+      Neptune
+
+To learn more about the ``OrderBy()`` method, see the `.NET API documentation
+<https://learn.microsoft.com/en-us/dotnet/api/system.linq.queryable.orderby>`__.
+
+.. tip:: Sort in Descending Order
+
+   You can sort the results of a query in descending order by using the
+   ``OrderByDescending()`` method.
+
+   To learn more about the ``OrderByDescending()`` method, see the `.NET API documentation
+   <https://learn.microsoft.com/en-us/dotnet/api/system.linq.queryable.orderbydescending>`__.
+
+You can perform a secondary sort on your query by using the ``ThenBy()`` method. The
+``ThenBy()`` method sorts the results of the ``OrderBy()`` method in ascending
+order based on a specified sort criteria. The ``ThenBy()`` method should be
+chained to the ``OrderBy()`` method.
```

Comment:
Might be worth mentioning ThenByDescending as well.

Reply:
Added a note further down about ThenByDescending